Assets/MixedRealityToolkit.Extensions/QRCodesTracker/Scripts/QRCodesManager.cs

```diff
@@ -0,0 +1,297 @@
+                        System.Numerics.Matrix4x4? relativePose = coordinateSystem.TryGetTransformTo(appSpatialCoordinateSystem);
+                        if (relativePose != null)
+                        {
+                            System.Numerics.Matrix4x4 newMatrix = relativePose.Value;
+
+                            // Platform coordinates are all right handed and unity uses left handed matrices. so we convert the matrix
+                            // from rhs-rhs to lhs-lhs
+                            // Convert from right to left coordinate system
+                            newMatrix.M13 = -newMatrix.M13;
+                            newMatrix.M23 = -newMatrix.M23;
+                            newMatrix.M43 = -newMatrix.M43;
+
+                            newMatrix.M31 = -newMatrix.M31;
+                            newMatrix.M32 = -newMatrix.M32;
+                            newMatrix.M34 = -newMatrix.M34;
+
+                            System.Numerics.Vector3 winrtScale;
+                            System.Numerics.Quaternion winrtRotation;
+                            System.Numerics.Vector3 winrtTranslation;
+                            System.Numerics.Matrix4x4.Decompose(newMatrix, out winrtScale, out winrtRotation, out winrtTranslation);
+
+                            var translation = new Vector3(winrtTranslation.X, winrtTranslation.Y, winrtTranslation.Z);
+                            var rotation = new Quaternion(winrtRotation.X, winrtRotation.Y, winrtRotation.Z, winrtRotation.W);
+                            location = Matrix4x4.TRS(translation, rotation, Vector3.one);
+
+                            return true;
+                        }
+                        else
+                        {
+                            Debug.LogWarning("QRCode location unknown or not yet available.");
+                            return false;
+                        }
+                    }
+                    else
+                    {
+                        Debug.LogWarning("Failed to obtain coordinate system for application");
+                        return false;
+                    }
+                }
+                catch(Exception e)
+                {
+                    Debug.LogWarning($"Note: TryGetLocationForQRCode needs to be called from main thread: {e}");
+                    return false;
+                }
+            }
+            else
+            {
+                Debug.LogWarning("Failed to obtain coordinate system for QRCode");
+                return false;
+            }
+        }
+#endif // WINDOWS_UWP
+
+        public Guid GetIdForQRCode(string qrCodeData)
+        {
+            lock (qrCodesList)
+            {
+                foreach (var ite in qrCodesList)
+                {
+                    if (ite.Value.Code == qrCodeData)
+                    {
+                        return ite.Key;
+                    }
+                }
+            }
+            return new Guid();
```

Comment:
>new Guid(); [](start = 19, length = 11)

Do you wish to return an all 0 guid in case of failure to lookup? If so, perhaps just use default(Guid)? Better pattern would be to do TryGet

Reply:
I will address qrcodesmanager changes in a separate review. This code is currently being worked on in a different repo.